downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -291,11 +291,12 @@ abstract class VersionControlSystem {
     open fun initWorkingTree(targetDir: File, vcs: VcsInfo): WorkingTree = getWorkingTree(targetDir)
 
     /**
-     * Update the [working tree][workingTree] by checking out the given [revision], optionally [recursively][recursive].
+     * Update the [working tree][workingTree] by checking out the given [revision], eventually limited to the given
```

Comment:
I think this should be "optionally" instead of "eventually". Also, shouldn't we set default values for optional parameters? For the path I wonder if allowing `null` for not using a sparse checkout would be better than using `""`.

Reply:
I didn't want to double "optionally" which is already used in the same sentence later, but I can reword that. Makes sense to use defaults for optional parameters, will do in a separate commit.

Reply:
Thanks, because "eventually" has a very different meaning than "optionally".